### DISCUSSION TITLE
How Do I Apply A Custom Font Or Change Font Size?

### DISCUSSION THREAD

**Discussion by Kazancik** [CosmosOS/Cosmos#3103] [Q&A] [answered]
I use DrawString to write on screen but I couldn't find a way to change font size or load a font, how do I do it?

**Comment by Gabolate** [comment]
To load another font you need it to be in the PSF (PC Screen Font) format, you can do something like this to use in DrawString:
```csharp
PCScreenFont myFont = PCScreenFont.LoadFont(yourFontBytes);
```

**Reply by Kazancik** [reply]
How do I change Font size?

**Reply by ADev531** [reply, marked as answer]
In PCScreenFont, Simply you can't.
You need to load a font with custom size, no option to change size.

**Comment by jason-pezzimenti** [comment]
I'm also trying to use a custom font (TrueType Font), but when I use:

```cs
		public byte[] defaultFontFile;
		public PCScreenFont DefaultFont;

		protected override void BeforeRun()
		{
			// Do initial loading.
			defaultFontFile = File.ReadAllBytes(Properties.Resources.Saira);

			if (defaultFontFile != null)
			{
				DefaultFont = PCScreenFont.LoadFont(defaultFontFile);
			}
		}
```

I get the following Build error:

1>IL2CPU : error : Exception: System.Exception: Native code encountered, plug required. Check build output for more information.
1>  DO NOT REPORT THIS AS A BUG.
1>  Please see http://www.gocosmos.org/docs/plugs/missing/

However, since I've only _just_ started with COSMOS, I clearly have no idea what to do next. Obviously a Plug is required. I don't know which one, or where to find one.

Is anybody able to help me out with this? :)

**Reply by Gabolate** [reply]
You can't use TrueType fonts directly in cosmos, you need them in the PSF format (PCScreenFont), there are some from the linux terminal iirc

**Reply by jason-pezzimenti** [reply]
Thanks.
